File: backend/app/models/automata_model.py
class AfndModel:
    def __init__(self, states, alphabet, transitions, initial, accepting):
        self.states = states
        self.alphabet = alphabet
        self.transitions = transitions
        self.initial = initial
        self.accepting = accepting

    def to_dict(self):
        return {
            'states': self.states,
            'alphabet': self.alphabet,
            'transitions': self.transitions,
            'initial': self.initial,
            'accepting': self.accepting
        }


class AfdModel:
    def __init__(self, states, alphabet, transitions, initial, accepting):
        self.states = states
        self.alphabet = alphabet
        self.transitions = transitions
        self.initial = initial
        self.accepting = accepting

    def to_dict(self):
        return {
            'states': self.states,
            'alphabet': self.alphabet,
            'transitions': self.transitions,
            'initial': self.initial,
            'accepting': self.accepting
        }

File: frontend/src/app/shared/models/automata.model.ts
export interface SimulationStep {
  symbol: string | null;
  states?: string[];
  state?: string | null;
}

export interface SimulationResult {
  input: string[];
  accepted: boolean;
  steps: SimulationStep[];
  final_states?: string[];
  final_state?: string | null;
  definition: AutomataDefinition;
}

export interface AutomataDefinition {
  states: string[];
  alphabet: string[];
  transitions: Record<string, Record<string, string[] | string>>;
  initial: string;
  accepting: string[];
}